Source/Session.swift

```diff
@@ -1058,6 +1063,7 @@ open class Session {
         guard !request.isCancelled else { return }
 
         guard let adapter = adapter(for: request) else {
+            guard shouldCreateTask() else { return }
```

Comment:
I don't think we should do this work twice. The check below should be enough, so do we want to remove this one?

Reply:
Ah I see why, sorry.

Reply:
It doesn't. This is inside the `guard` for when we don't have an `adapter`.